docs: storage metadata,info,exists (#29179)
supabase/supabase: apps/docs/content/guides/storage/management/metadata.mdx

---
id: 'storage-management'
title: 'Object Metadata'
description: 'Learn how to store and retrieve object metadata'
subtitle: 'Learn how to get object metadata'
sidebar_label: 'Object Metadata'
---

## Object Metadata
Each object in a bucket can have metadata associated with it. Metadata is a set of key-value pairs that describe the object.
You can use metadata to store information such as the object's content type, author, or any other custom data.

> 

Reviewer: ```suggestion
```

Each upload method in the Storage API accepts a `metadata` option that allows you to add custom metadata to the object.

> Please refer to the followings guides for more information on how to upload objects with metadata.:

Reviewer: ```suggestion
See the following guides for more information on how to upload objects with metadata:
```

- [Standard Uploads](/docs/guides/storage/uploads/standard-uploads#metadata)
- [Resumable Upload](/docs/guides/storage/uploads/resumable-uploads#metadata)

<Admonition type="note">

    The metadata is immutable, meaning that once an object is uploaded with metadata, you cannot update the metadata.
    You will currently need to update objects by re-uploading them with the new metadata.

</Admonition>

## Retrieving Metadata

> To retrieve the metadata of an object, using the `info()` method.

Reviewer: ```suggestion
To retrieve the metadata of an object, use the `info()` method.
```
